lib/web_ui/test/golden_tests/engine/canvas_draw_image_golden_test.dart

```diff
@@ -0,0 +1,136 @@
+    await webOnlyInitializePlatform();
+    webOnlyFontCollection.debugRegisterTestFonts();
+    await webOnlyFontCollection.ensureFontsLoaded();
+  });
+
+  test('Paints image', () async {
+    final RecordingCanvas rc =
+    RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
+    rc.save();
+    rc.drawImage(createTestImage(), Offset(0, 0), new Paint());
+    await _checkScreenshot(rc, 'draw_image');
+  });
+
+  test('Paints image with transform', () async {
+    final RecordingCanvas rc =
+    RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
+    rc.save();
+    rc.translate(50.0, 100.0);
+    rc.rotate(math.pi / 4.0);
+    rc.drawImage(createTestImage(), Offset(0, 0), new Paint());
+    await _checkScreenshot(rc, 'draw_image_with_transform');
+  });
+
+  test('Paints image with transform and offset', () async {
+    final RecordingCanvas rc =
+    RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
+    rc.save();
+    rc.translate(50.0, 100.0);
+    rc.rotate(math.pi / 4.0);
+    rc.drawImage(createTestImage(), Offset(30, 20), new Paint());
+    await _checkScreenshot(rc, 'draw_image_with_transform_and_offset');
+  });
+
+  test('Paints image with transform using destination', () async {
+    final RecordingCanvas rc =
+    RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
+    rc.save();
+    rc.translate(50.0, 100.0);
+    rc.rotate(math.pi / 4.0);
+    Image testImage = createTestImage();
+    double testWidth = testImage.width.toDouble();
+    double testHeight = testImage.height.toDouble();
+    rc.drawImageRect(testImage, Rect.fromLTRB(0, 0, testWidth, testHeight),
+        Rect.fromLTRB(100, 30, 2 * testWidth, 2 * testHeight), new Paint());
+    await _checkScreenshot(rc, 'draw_image_rect_with_transform');
+  });
+
+  test('Paints image with source and destination', () async {
+    final RecordingCanvas rc =
+    RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
+    rc.save();
+    Image testImage = createTestImage();
+    double testWidth = testImage.width.toDouble();
+    double testHeight = testImage.height.toDouble();
+    rc.drawImageRect(testImage, Rect.fromLTRB(testWidth / 2, 0, testWidth, testHeight),
+        Rect.fromLTRB(100, 30, 2 * testWidth, 2 * testHeight), new Paint());
+    await _checkScreenshot(rc, 'draw_image_rect_with_source');
+  });
+
+  test('Paints image with transform using source and destination', () async {
+    final RecordingCanvas rc =
+    RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
+    rc.save();
+    rc.translate(50.0, 100.0);
+    rc.rotate(math.pi / 6.0);
+    Image testImage = createTestImage();
+    double testWidth = testImage.width.toDouble();
+    double testHeight = testImage.height.toDouble();
+    rc.drawImageRect(testImage, Rect.fromLTRB(testWidth / 2, 0, testWidth, testHeight),
+        Rect.fromLTRB(100, 30, 2 * testWidth, 2 * testHeight), new Paint());
+    await _checkScreenshot(rc, 'draw_image_rect_with_transform_source');
+  });
```

Comment:
I think tests with clips would be very useful too, since that part is a little tricky in the implementations. Particular the cases when two images are clipped by a single clip, or when an image and a paragraph are both clipped.

Reply:
The transform with src test clips before rendering off center. It actually uncovered a bug when compared to drawImageRect screenshot that was taken with prior canvas drawImageScaled call.